packages/mds-event-processor/src/proc-event.ts

```diff
@@ -0,0 +1,271 @@
+import stream from '@mds-core/mds-stream'
+
+import { getAnnotationData, getAnnotationVersion } from './annotation'
+import {
+  CE_TYPE,
+  StateEntry,
+  TripEvent,
+  TripTelemetry,
+  EVENT_STATUS_MAP,
+  VEHICLE_EVENT,
+  VEHICLE_REASON
+} from '@mds-core/mds-types'
+
+/*
+    Event processor that runs inside a Kubernetes pod.
+    Streams cloudevents from mds agency and process them in multiple ways:
+
+        1) quality checks
+        2) status changes
+        3) trip identification
+
+    Processed events/telemetry are added to various caches keyed as follows:
+
+        1) device:state (latest event/telemetry for a device)
+        2) trip:events (events linked to trips of a device)
+        3) device:ID:trips (all telemetry linked to all trips of a device)
+            - ID is the combination 'provider_id:device_id'
+
+    A Postgres table is also populated to store historical states:
+
+        REPORTS_DEVICE_STATES:
+          PRIMARY KEY = (provider_id, device_id, timestamp, type)
+          VALUES = deviceState
+*/
+
+async function eventHandler() {
+  await dataHandler('event', async function(type: CE_TYPE, data: any) {
+    console.log(type, data)
+    return processRaw(type, data)
+  })
+}
+
+async function processRaw(type: CE_TYPE, data: any) {
+  const { timestamp, device_id, provider_id, recorded } = data as {
+    timestamp: number
+    device_id: string
+    provider_id: string
+    recorded: number
+  }
+  const lastState = JSON.parse(await cache.hget('device:state', provider_id + ':' + device_id)) as StateEntry
+  // Construct state
+  const baseDeviceState = {
+    vehicle_type: 'scooter',
+    type: type.substring(type.lastIndexOf('.') + 1),
+    timestamp: timestamp,
+    device_id: device_id,
+    provider_id: provider_id,
+    time_recorded: recorded,
+    annotation_version: getAnnotationVersion()
+  } as StateEntry
+
+  switch (baseDeviceState.type) {
+    case 'event': {
+      const { event_type, telemetry, event_type_reason, trip_id, service_area_id } = data as {
+        event_type: VEHICLE_EVENT
+        telemetry: any
+        event_type_reason: VEHICLE_REASON
+        trip_id: string
+        service_area_id: string
+      }
+      const gps = telemetry ? telemetry.gps : null
+      const charge = telemetry ? telemetry.charge : null
+      const annotation = gps ? getAnnotationData(gps) : null
+      //TODO decide what to do with annotation data
+      const district = annotation ? (annotation.areas.length ? annotation.areas[0].id : null) : null
+      const deviceState = {
+        ...baseDeviceState,
+        annotation: annotation,
+        gps: gps,
+        service_area_id: service_area_id,
+        charge: charge,
+        state: EVENT_STATUS_MAP[event_type as VEHICLE_EVENT],
+        event_type: event_type,
+        event_type_reason: event_type_reason,
+        trip_id: trip_id
+      } as StateEntry
+      // Take necessary steps on event trasitions
+      switch (data.event_type) {
```

Comment:
For brevity's sake, something like `['trip_start', 'trip_enter', 'trip_leave', 'trip_end'].find(trip_type => data.event_type === trip_type)` would be better.